src/ddp/interface.ts

```diff
@@ -0,0 +1,11 @@
+import type { KeyofToArray } from '@murongg/utils/index'
+import type { ProofParams } from '../types'
+import { dspParamsNormalize } from '../common/utils'
+
+export abstract class DataDestinationPlugin<GP extends object> {
+  abstract goParams: KeyofToArray<GP>
+  toGoParams(params: Record<string, any>) {
+    return dspParamsNormalize(this.goParams as string[], params) as GP
+  }
+  abstract go(cleId: string, proofParams: ProofParams, goParams: GP): Promise<void >
```

Comment:
Is `go` really a good name for our case?

Reply:
it's like "go to destination". but do you have other suggestions?

Reply:
merging, we can still rename this later, let's keep this conv until decide